tf-modules/data-persistence/elasticsearch.tf

```
@@ -0,0 +1,118 @@
+locals {
+  inside_vpc         = length(var.subnet_ids) > 0 ? true : false
+  deploy_inside_vpc  = var.include_elasticsearch && local.inside_vpc
+  deploy_outside_vpc = var.include_elasticsearch && local.inside_vpc == false
+  include_es_policy  = length(var.es_trusted_role_arns) > 0 ? true : false
+  es_domain_name     = "${var.prefix}-${var.elasticsearch_config.domain_name}${local.inside_vpc ? "-vpc" : ""}"
+}
+
+data "aws_region" "current" {}
+
+data "aws_caller_identity" "current" {}
+
+data "aws_iam_policy_document" "es_access_policy" {
+  statement {
+    actions = [
+      "es:*"
+    ]
+
+    principals {
+      type        = "AWS"
+      identifiers =  distinct(compact(var.es_trusted_role_arns))
+    }
+
+    resources = [
+      "arn:aws:es:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:domain/${local.es_domain_name}/*"
```

Would it make sense to reference this based on the conditionals/vars that define which ES domain is created? Something like:
```suggestion
"${local.inside_vpc ? aws_elasticsearch_domain.es_vpc.arn : aws_elasticsearch_domain.es.arn}/*"
```
Or does that break under certain conditions?

---

@marc @nemreid I can't make the `es_access_policy` resource reference the `aws_elasticsearch_domain` ARN because it is used in `es_domain_policy` resource, which itself depends on the `aws_elasticsearch_domain` resource. And the `es_domain_policy` resource has to depend on the `aws_elasticsearch_domain` resource otherwise it tries to create the domain policy first and complains that the referenced domain doesn't exist. Let me know if that doesn't make sense. I can also double check

---

Stupid circular dependencies.

---

Is this circular? It sounds like the dependency tree is (in order of creation) `aws_elasticsearch_domain <- es_access_policy <- es_domain_policy`. I'm not sure I understand the issue there.

---

Hmm, yeah. Let me look at this again